x/incentives/keeper/grpc_query_test.go

```diff
@@ -109,6 +109,39 @@ func (suite *KeeperTestSuite) TestGRPCActiveGauges() {
 	suite.Require().Equal(res.Data[0].String(), expectedGauge.String())
 }
 
+func (suite *KeeperTestSuite) TestGRPCActiveGaugesPerDenom() {
+	suite.SetupTest()
+
+	// initial check
+	res, err := suite.querier.ActiveGaugesPerDenom(sdk.WrapSDKContext(suite.ctx), &types.ActiveGaugesPerDenomRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 0)
+
+	// create a gauge
+	gaugeID, _, coins, startTime := suite.SetupNewGauge(false, sdk.Coins{sdk.NewInt64Coin("stake", 10)})
+
+	// final check
+	res, err = suite.querier.ActiveGaugesPerDenom(sdk.WrapSDKContext(suite.ctx), &types.ActiveGaugesPerDenomRequest{})
+	suite.Require().NoError(err)
+	suite.Require().Len(res.Data, 1)
```

Comment:
This is failing right now, which is actually correct behavior at the moment. This is because gauges only move from upcoming -> active during the epoch logic at the moment. (We should change the state machine to make it possible for gauges to start out as active, not always as upcoming)

You should call `k.BeginDistribution(suite,ctx, gauge)` before this line to make it work. (Also define the gauge return value from `suite.SetupNewGauge`

We should do the same thing to fix the UpcomingGaugesPerDenom test